### PR TITLE
Move link handling code to OsuGame

### PR DESCRIPTION
**Pull request by peppy**
Split out of https://github.com/ppy/osu/pull/5574.

Added protection for potentially being called before the game / gamehost / target overlay are in a correct state.

### CODE REVIEW

**Review by bdach — DISMISSED** [review]
I think this makes the code look better overall. Shame it makes `OsuGame` grow in size a little, but not much that can be done about that I guess.

**Comment by peppy** [comment]
If we get to the point we want to move this out, a `static class LinkRouter { void Route(string link, OsuGame game) { ... }` could work, since the target functions are all `public`

File: osu.Game/OsuGame.cs [hunk 1]
```diff
@@ -215,31 +215,102 @@ private void load(FrameworkConfigManager frameworkConfig)
 
         private ExternalLinkOpener externalLinkOpener;
 
-        public void OpenUrlExternally(string url)
+        /// <summary>
+        /// Handle an arbitrary URL. Displays via in-game overlays where possible.
+        /// This can be called from a non-thread-safe non-game-loaded state.
+        /// </summary>
+        /// <param name="url">The URL to load.</param>
+        public void HandleLink(string url) => Schedule(() => HandleLink(MessageFormatter.GetLinkDetails(url)));
```
Comment (smoogipoo) [inline osu.Game/OsuGame.cs]: The schedule should exist on the other method. This also means that the other method isn't actually thread-safe.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/OsuGame.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/OsuGame.cs]: Fair call

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**